pkg/ostreeuploader/push.go

```diff
@@ -103,6 +115,32 @@ var (
 	}
 )
 
+func (t Token) SetAuthHeader(req *http.Request) {
+	if t.Schema == OAuth2Token {
+		req.Header.Set("Authorization", "Bearer "+t.Value)
+	} else if t.Schema == FioToken {
+		req.Header.Set("osf-token", t.Value)
+	} else {
+		log.Fatalf("Unsupported type of token: %s\n", t.Schema)
+	}
+}
```

Comment:
you could probably make this type safe by doing something like:
```
type FioAuthHandler interface {
    SetAuthHeader(req *http.Request) error
}

type TokenAuth struct {
  Token: string
}

func (t TokenAuth) SetAuthHeader ....
// repeat for Oauth2 helper
```

Reply:
Not sure what exactly you meant since I see a few ways to accomplish "type safety". I implemented one of the ways which I believe is the most minimalistic.

Reply:
what you have is perfect. it eliminated the if/else-fail